superset/db_engine_specs/base.py

```diff
@@ -1410,7 +1410,8 @@ def build_sqlalchemy_uri(
         parameters: BasicParametersType,
         encryted_extra: Optional[Dict[str, str]] = None,
     ) -> str:
-        query = parameters.get("query", {})
+        # make a copy so that we don't update the original
+        query = dict(parameters.get("query", {}))
```

Comment:
Nit to be more explicit:

```suggestion
        query = parameters.get("query", {}).copy()
```